src/apps/btc/btc_sign.c

```diff
@@ -60,7 +60,8 @@ static uint64_t _outputs_sum_ours = 0;
 static uint64_t _outputs_sum_out = 0;
 
 // used during STATE_INPUTS_PASS1
-static sha256_context_t _hash_prevouts_ctx, _hash_sequence_ctx;
+static void* _hash_prevouts_ctx = NULL;
+static void* _hash_sequence_ctx = NULL;
```

Comment:
Why not `rust_sha256_new();` instead of `NULL`? I see these are initialized in `_reset()` and the latter is called from `app_btc_sign_init` but `app_btc_sign_input` is also invoked from the commander when `Request_btc_sign_input_tag` is sent. If the host is misbehaving, could that lead to unnecessary panics/crashes?

Reply:
It's a static var, so it needs to be constant. 

reset is called at the start of each transaction signing, when it is aborted at any time, or after it finishes normally. I don't see a way for the host to invoke a crash, do you?

`rust_sha256_free()` can be called on NULL without any effect.

Reply:
Oh right. I missed that `app_btc_sign_input` checks what `_state` it's in on https://github.com/digitalbitbox/bitbox02-firmware/blob/4a2cac951381032ac0fcb511d0ea747c478c95b5/src/apps/btc/btc_sign.c#L375-L377.
Never mind. :+1: